src/modules/Dashboard/features/RespondentData/RespondentDataSummary/ReportMenu/ReportMenu.tsx

```diff
@@ -1,42 +1,72 @@
-import { useContext } from 'react';
-import { Box } from '@mui/material';
+import { useContext, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import {
-  theme,
-  StyledHeadlineLarge,
-  StyledLabelLarge,
   StyledBodyLarge,
   variables,
+  StyledStickyHeadline,
+  theme,
+  StyledStickyHeader,
 } from 'shared/styles';
 import { RespondentDataContext } from 'modules/Dashboard/pages/RespondentData/RespondentData.context';
-import { useRespondentLabel } from 'shared/hooks';
+import {
+  OFFSET_TO_SET_STICKY,
+  OFFSET_TO_UNSET_STICKY,
+  useHeaderSticky,
+  useRespondentLabel,
+} from 'shared/hooks';
+import { ContentWithTooltip, DataTableItem } from 'shared/components';
 
 import { StyledMenu } from '../../RespondentData.styles';
-import { StyledActivity } from './ReportMenu.styles';
+import { StyleContainer, StyledActivity, StyledHeaderContainer } from './ReportMenu.styles';
 import { ReportMenuProps } from './ReportMenu.types';
 
 export const ReportMenu = ({ activities }: ReportMenuProps) => {
   const { t } = useTranslation();
   const respondentLabel = useRespondentLabel();
   const { selectedActivity, setSelectedActivity } = useContext(RespondentDataContext);
+  const containerRef = useRef<HTMLElement | null>(null);
+  const isHeaderSticky = useHeaderSticky(
+    containerRef,
+    OFFSET_TO_SET_STICKY / 2,
+    OFFSET_TO_UNSET_STICKY / 2,
+  );
 
   return (
-    <StyledMenu data-testid="report-menu">
-      <Box sx={{ margin: theme.spacing(0, 2.4, 3.2) }}>
-        <StyledHeadlineLarge>{t('activities')}</StyledHeadlineLarge>
-        <StyledLabelLarge>{respondentLabel}</StyledLabelLarge>
-      </Box>
-      {activities?.map((activity, index) => (
-        <StyledActivity
-          key={String(activity.id)}
-          isSelected={selectedActivity?.id === activity.id}
-          onClick={() => setSelectedActivity(activity)}
-          data-testid={`respondents-summary-activity-${index}`}
-        >
-          <StyledBodyLarge color={variables.palette.on_surface}>{activity.name}</StyledBodyLarge>
-        </StyledActivity>
-      ))}
+    <StyledMenu ref={containerRef} data-testid="report-menu" sx={{ p: 0 }}>
+      <StyledStickyHeader
```

Comment:
Could you move this header to its own component, where ``useHeaderSticky`` is launched? This will prevent the whole ``ReportMenu`` from re-rendering every time ``isHeaderSticky`` changes.